packages/page-explorer/src/BlockInfo/ByHash.tsx

```diff
@@ -110,13 +112,33 @@ function BlockByHash ({ className = '', error, value }: Props): React.ReactEleme
   const parentHash = getHeader?.parentHash.toHex();
   const hasParent = !getHeader?.parentHash.isEmpty;
 
+  const isFinalized = useMemo(() => {
+    if (bestNumberFinalized && blockNumber) {
+      return bestNumberFinalized.toNumber() >= blockNumber.toNumber();
```

Comment:
how this would work with blocks from fork? Wouldnt all of them be marked as a finalized if they blocknumber would be less than last finalized?

Reply:
`bestNumberFinalized` just shows result of rpc call `rpc.chain.finalizedHead` , so this is just a view of what a given validator thinks finalized chain is. So it depends which validator wallet connects to.

Reply:
We can't also check for existence of justification, as wallet does not have access to all justifications. So probably to the `isFinalized` check we would also have to add a check whether specific block is in the best chain for us.

Reply:
You're right. Since blocks don't include the information about the fork they're on (like a session-id or sth which could be compared to _verified/trusted_ ID for equivalence check or dependency) we can't infer that knowledge from the block alone. Even having justifications wouldn't solve the problem entirely as justifications can be forged so we would need to know which validator set to trust.

Reply:
Okay, I think I see the problem now. But the problem is not as severe as before, given that we limit block production now to N blocks. Though still, it can be M forks of length as much as N blocks. So to be on the safe side, we should check belonging to the finalized chain when marking the block as finalized as you suggested. Let me think how this can be done. Thankfully `rpc.chain.getFinalizedHead` returns hash, not block number and we did that change recently to limit block production if not finalizing. Otherwise this PR might be not doable.

Reply:
You will have to traverse, starting from `rpc.chain.getFinalizedHead` using parent hashes to see if the block we have belongs to that chain. Thankfully, we know heights of both (or can find them) so we have to check up to `diff` blocks (in our case `N`). Shouldn't be difficult to implement, just not the most effective if we don't cache the chain on the client side.

Reply:
this might be super slow in the long run - we might think about adding following rpc API: `is_finalized(block_hash) -> bool`. In the node it should be super easy to check if the block was finalized :)

Reply:
UNfortunately I did not see any such rpc in the node API :(

Reply:
Of course but that would require upgrade & release but doing it in the wallet does not :)

Reply:
yep, I think we should use the traverse solution for now, and add rpc in later release

Reply:
But what if someone wants to check a block that is far in the past. Asking for 1 milion blocks using RPC does not sound good.

What I would suggest is ask for the the block with `number + 1`, if it is `<= finalized head number`. If we ask for number it should return a block from best chain. Then we check its parent. If it is the block we are investigating, we can use the same check here and it should be fine.

Reply:
if you go to https://azero.dev/#/explorer/forks then sometimes it takes 5-6 blocks before the first one is recognized as finalized.

Reply:
The block you're looking at may be more than just 1 block ahead of the last finalized one.

Reply:
Then it is not finalized from perspective of wallet, right? Same thing happens with the other check in this PR.

Reply:
Or even same number. Should also be fine

Reply:
I've extracted common code to a hook, and added an additional check that uses `rpc.chain.getBlockNumber`, which returns block hash from the **canonical** chain. That should do the job.